packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -502,13 +505,36 @@ export class MultiSinkLogger extends TelemetryLogger {
 		this.loggers = loggers;
 	}
 
+	// This is minimum of minLlogLevel of all loggers.
+	private _minLogLevelOfAllLoggers: LogLevel | undefined;
+
+	public get minLogLevel(): LogLevel | undefined {
+		if (this._minLogLevelOfAllLoggers === undefined) {
+			const logLevels: number[] = [];
+			for (const logger of this.loggers) {
+				logLevels.push(logger.minLogLevel ?? LogLevel.default);
+			}
+			this._minLogLevelOfAllLoggers = Math.min(...logLevels);
+		}
+		// This will be eual to LogLevel.Default in case none of the loggers have the minLogLevel defined.
+		return this._minLogLevelOfAllLoggers;
+	}
```

Comment:
Would it be easier to initialize `_minLogLevelOfAllLoggers` to the default log level and just return it here? Then the only place where it needs to be checked/updated is on `addLogger` and we can remove its `undefined` check from the condition there.

The way we set the value for `_minLogLevelOfAllLoggers` here and in `addLogger()` feels like it could have weird edge case bugs.

Reply:
It should first be decided based on the log level of all loggers supplied in the constructor. We can not initialize with default as what if the minLogLevel of all loggers is LogLevel.error.

Reply:
Fair, but then that logic fits better in the constructor than here, I think. I was particularly interested in the case where the constructor receives no loggers and they're all added later (which I think is possible? Not 100% sure).

Reply:
fixed and added test.